### PR TITLE
Adds support for fonts in overlay

### PR DESCRIPTION
**Pull request by inderpreetsingh**
Adds multiple font options in bani overlay window.
Screenshot 2020-03-25 at 1 06 44 AM

### CODE REVIEW

**Comment by maneetpaul** [comment]
@inderpreetsingh is the window size reactive? Meaning, if the window is big enough, will the layout options appear on the same row as the other options?

File: www/js/overlay.js [hunk 1]
```diff
@@ -176,6 +176,46 @@ const resizeButtonFactory = (increaseFunc, decreaseFunc) =>
...
+window.addEventListener('click', e => {
+  if (!e.target.classList.contains('select-value')) {
+    closeAllDropDowns();
+  } else {
+    toggleDropDown(e.target.parentElement);
+  }
+});
+
+const fontListFactory = (list, propName) => {
+  const lowerize = str => str.replace(/ /g, '').toLowerCase();
```
Comment (tsingh777) [inline www/js/overlay.js]: should probably define this outside of factory function.

File: www/src/scss/obs/styles.scss [hunk 2]
```diff
@@ -1,3 +1,45 @@
...
+}
+
+@font-face {
+  font-family: 'caveat';
+  src: url('english/Caveat.ttf') format('truetype');
+}
+
+@font-face {
+  font-family: 'ptserif';
+  src: url('english/PT-Serif.ttf') format('truetype');
```
Comment (tsingh777) [inline www/src/scss/obs/styles.scss]: think this english folder is missing

File: www/js/overlay.js [hunk 3]
```diff
@@ -176,6 +176,46 @@ const resizeButtonFactory = (increaseFunc, decreaseFunc) =>
...
+  const onClickOption = (e, font) => {
+    e.target.parentElement.parentElement.querySelector('.select-value').innerHTML = font;
+    overlayVars[propName] = lowerize(font);
+    savePrefs();
+  };
+  const options = list.map(font => h(`div.option`, { onclick: e => onClickOption(e, font) }, font));
+  return h(
+    'div.custom-select',
```
Comment (tsingh777) [inline www/js/overlay.js]: why didn't we just use a select element like we have in other parts of the app?

File: www/js/defaults.json [hunk 4]
```diff
@@ -91,6 +104,8 @@
         "gurbaniTextColor": "#EAC445",
         "overlayLogo": true,
         "overlayLarivaar": false,
+        "translationFont": "muli",
+        "gurbaniFont": "gurbaniakhar",
```
Comment (tsingh777) [inline www/js/defaults.json]: why not just keep it the same as font name? that would remove need to transform the value when reading/writing.

File: www/js/overlay.js [hunk 5]
```diff
@@ -176,6 +176,46 @@ const resizeButtonFactory = (increaseFunc, decreaseFunc) =>
...
+  if (!e.target.classList.contains('select-value')) {
+    closeAllDropDowns();
+  } else {
+    toggleDropDown(e.target.parentElement);
+  }
+});
+
+const fontListFactory = (list, propName) => {
+  const lowerize = str => str.replace(/ /g, '').toLowerCase();
+  const onClickOption = (e, font) => {
```
Comment (tsingh777) [inline www/js/overlay.js]: define outside of factory function.

File: www/js/defaults.json (hunk 4 above)
Comment (tsingh777) [reply www/js/defaults.json]: Just wondering but can the bold ones be their own font family? The key is arbitrary so would it makes it easier for each option to correspond to its own font-face?

File: www/obs/index.html [hunk 6]
```diff
@@ -89,6 +89,23 @@
         document.querySelector('.o-gurbani').style.color =
           overlayPrefs.overlayVars.gurbaniTextColor;
 
+        document.body.classList.remove(
```
Comment (tsingh777) [inline www/obs/index.html]: is this duplicate? the same is done on line 109

File: www/js/overlay.js (hunk 3 above)
Comment (inderpreetsingh) [reply www/js/overlay.js]: It wasn't looking as good here - and also needed some functionality which allows me to open the dropdown as drop-up depending on window size.

**Comment by inderpreetsingh** [comment]
@maneetpaul yes it is now. :)

File: www/js/overlay.js [hunk 7]
```diff
@@ -176,6 +176,48 @@ const resizeButtonFactory = (increaseFunc, decreaseFunc) =>
...
+  overlayVars[propName] = font;
+  savePrefs();
+};
+
+const fontListFactory = (list, propName) => {
+  const options = list.map(font =>
+    h(`div.option`, { onclick: e => fontSwitch(e, font, propName) }, font),
+  );
+  return h(
+    'div.custom-select',
```
Comment (tsingh777) [inline www/js/overlay.js]: could have probably extracted this component to separate file. can follow up on this later.

**Pull request merged**